src/test/java/se/iths/springbootgroupproject/controllers/GuestControllerTest.java

```diff
@@ -0,0 +1,96 @@
+package se.iths.springbootgroupproject.controllers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import se.iths.springbootgroupproject.dto.MessageAndUsername;
+import se.iths.springbootgroupproject.entities.User;
+import se.iths.springbootgroupproject.services.MessageService;
+import se.iths.springbootgroupproject.services.github.GithubOAuth2UserService;
+
+import java.time.LocalDate;
+import java.util.List;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(controllers = GuestController.class)
+class GuestControllerTest {
+
+    @Autowired
+    MockMvc mockMvc;
+
+    @Autowired
+    ObjectMapper mapper;
+
+    @MockBean
+    MessageService messageService;
+
+    @MockBean
+    private GithubOAuth2UserService githubOAuth2UserService;
```

Comment:
This seems to not be in use. Could be removed.

Reply:
It has something to do with the application context. It is a dependency in security config and when it wasnt present in the controller test class the whole testclass would just turn into errors. It confuses me though that it looks like it is needed, but never is user.. It might be something that @MockMvc needs to start the slice but not really sure. But for the tests to run it has to be there, atleast for me.

Reply:
I did not think of testing that in Guestcontroller, good catch! I will push a fix.